### PR TITLE
feat(query): add aggregation functions

### PR DESCRIPTION
**Pull request by KutsenkoA**

### CODE REVIEW

File: src/internal/query.ts [hunk 1]
```diff
@@ -160,12 +164,12 @@ export class Query<T = any> {
    * @param {IAggField} agg an aggregation object
    * @returns {string} compiled aggregation function
```
Comment (marlosin) [inline src/internal/query.ts]: Remove annotation type, `compodoc` will get from return type

File: src/internal/query.ts [hunk 2]
```diff
@@ -160,12 +164,12 @@ export class Query<T = any> {
...
    */
-  private compileAggregation(agg: IAggField): string {
-    if (agg.fn === "COUNT" && agg.col === "*") {
+  private compileAggregation(agg: IAggField): string | ICompiledAggField {
```
Comment (marlosin) [inline src/internal/query.ts]: The return type here would be always `ICompiledAggField` right?

File: src/internal/query.ts [hunk 3]
```diff
@@ -17,10 +17,14 @@ type KeyOfObject<T> = Extract<keyof T, string>;
  * <T> - generic resource object
  */
 export interface IAggField<T = any> {
-  fn: "COUNT" | "MIN" | "MAX" | "AVG" | "SUM" | "EVERY";
+  fn: "count" | "min" | "max" | "avg" | "sum";
```
Comment (marlosin) [inline src/internal/query.ts]: what do you think about providing this as an `enum` or even provide some functions for each that could be used like 
``` js
posts.select().fields("author", max("likes"))

File: e2e/stories/dataset/aggregations.e2e.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  it("should select max value", (done) => {
+    dom.dataset(DEFAULT_DATASET.NAME)
+      .select()
+      .fields({ fn: "max", col: "age"})
+      .subscribe({
+        next: (result) => {
+          expect(result).toEqual([{ max: 42 }]);
+          done();
+        },
+        error: done,
```
Comment (marlosin) [inline e2e/stories/dataset/aggregations.e2e.ts]: `error: done.fail`

File: e2e/stories/dataset/aggregations.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    await init(DEFAULT_DATASET.NAME, fields);
+    await dom.dataset(DEFAULT_DATASET.NAME).insert(data).execute();
+  });
+
+  afterAll(async () => cleaning());
+
+  it("should select max value", (done) => {
+    dom.dataset(DEFAULT_DATASET.NAME)
+      .select()
+      .fields({ fn: "max", col: "age"})
```
Comment (marlosin) [inline e2e/stories/dataset/aggregations.e2e.ts]: missing whitespace before `}`

File: src/internal/query.ts (hunk 1 above)
Comment (KutsenkoA) [reply src/internal/query.ts]: 👍

File: src/internal/query.ts (hunk 2 above)
Comment (KutsenkoA) [reply src/internal/query.ts]: right 👍

File: src/internal/query.ts (hunk 3 above)
Comment (KutsenkoA) [reply src/internal/query.ts]: I think very positively about this and I also have a plan to simplify the SDK (there is a card about this), where I'm going to introduce `jexia.min()`, `jexia.max()` and others. These function will return the same `IAggField` object, so this code will not be changed

File: e2e/stories/dataset/aggregations.e2e.ts (hunk 4 above)
Comment (KutsenkoA) [reply e2e/stories/dataset/aggregations.e2e.ts]: Hm, actually doesn't matter because `done` with any not empty argument will fail the test

**Pull request merged**